### PR TITLE
Install VTK from source on non-x86_64 systems

### PR DESCRIPTION
**Pull request by dacreman**
The VTK library is pip installed as a dependency of firedrake. However it is distributed as pre-built binary wheels for x86_64 systems only.  This PR adds a build_and_install_vtk function which builds VTK from source if the architecture is not x86_64 (otherwise the function does nothing). 

This change has been tested on Isambard (the GW4 ARM64 system) and includes logic for handling ARM64 and Cray specific options to the VTK build.

### CODE REVIEW

File: scripts/firedrake-install [hunk 1]
```diff
@@ -1006,6 +1006,114 @@ def build_and_install_slepc():
...
+    version="8.2.0",
+    author="Kitware",
+    author_email="kitware@kitware.com",
+    description="The Visualization Toolkit (VTK)",
+    long_description="The Visualization Toolkit (VTK)",
+    long_description_content_type="text/markdown",
+    url="https://vtk.org",
+    packages=setuptools.find_packages(),
+    classifiers=[
+        "Operating System :: ARM64",
```
Comment (dham) [inline scripts/firedrake-install]: Above we only check for non x86_64, but now we assume that this automatically means ARM. Is that really safe?

File: scripts/firedrake-install [hunk 2]
```diff
@@ -1006,6 +1006,114 @@ def build_and_install_slepc():
         log.info("No need to rebuild slepc4py")
 
 
+def build_and_install_vtk():
```
Comment (dham) [inline scripts/firedrake-install]: I'm a bit nervous about having this whole package definition in firedrake-install. Would it be better to have a git repo containing a Python package which does this instead.

**Comment by wence-** [comment]
Hi @dacreman. As discussed on Monday, I think it makes more sense if we don't put this code in firedrake-install, but instead split it out to build the wheels we need. We can then host those wheels somewhere and just point firedrake-install at them as appropriate.

**Comment by wence-** [comment]
@dacreman do you have any idea if you'll have time to make the changes we talked about?

**Comment by wence-** [comment]
This was fixed in #1672 by providing custom wheels in these cases.

**Pull request closed without merging**